File: 611. Valid Triangle Number/triangleNumber.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        int sz = nums.size();
        int ans = 0;

        sort(nums.begin(), nums.end());

        for(int i=sz-1; i>=2; i--){
            int left = 0;
            int right = i-1;

            while(left < right){
                int sum = nums[left] + nums[right];
                
                if(sum > nums[i]){
                    ans += (right- left);
                    right--;
                }
                else{
                    left++;
                }
            }
        }

        return ans;
    }
};

File: 611. Valid Triangle Number/triangleNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var triangleNumber = function (nums) {
    const len = nums.length
    nums.sort((a, b) => a - b)
    let ans = 0

    for (let i = len - 1; i >= 2; i--) {

        let left = 0
        let right = i - 1

        while (left < right) {

            if (nums[right] + nums[left] > nums[i]) {
                ans += right - left
                right--
            }
            else {
                left++
            }
        }
    }

    return ans
};